libvast/src/io/write.cpp

```diff
@@ -0,0 +1,33 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/io/write.hpp"
+
+#include "vast/error.hpp"
+#include "vast/filesystem.hpp"
+
+namespace vast::io {
+
+caf::error write(const path& filename, span<const byte> xs) {
+  file f{filename};
+  if (!f.open(file::write_only))
+    return make_error(ec::filesystem_error, "failed open file");
+  size_t bytes_written;
+  if (!f.write(xs.data(), xs.size(), &bytes_written))
+    return make_error(ec::filesystem_error, "failed to write chunk");
+  if (bytes_written != xs.size())
+    return make_error(ec::filesystem_error, "incomplete write");
```

Comment:
In case of an incomplete write, should we try to remove the written bytes?

Reply:
This may not be possible. The call site could remove the corrupted file, though.